frontend/pages/staking.tsx

```diff
@@ -40,20 +41,32 @@ const Staking: NextPage = () => {
     multipleStakeAccountsModalOption,
     setMultipleStakeAccountsModalOption,
   ] = useState<StakeAccount>()
-  const { data: stakeAccounts } = useStakeAccounts()
-  const [mainStakeAccount, setMainStakeAccount] = useState<StakeAccount>()
+
+  const wallet = useAnchorWallet()
+  const isWalletConnected = wallet !== undefined
+
+  const { isLoading: isStakeConnectionLoading } = useStakeConnection()
+  const { data: stakeAccounts, isLoading: isStakeAccountsLoading } =
+    useStakeAccounts()
+
+  // if things are loading, mainStakeAccount is undefined
+  // else if there are no previous stakeAccount, mainStakeAccount is null
+  // else mainStakeAccount is defined
+  const [mainStakeAccount, setMainStakeAccount] = useState<
```

Comment:
this `undefined | null` convention is going to be confusing and will trip people up in the future. If you really want to do something like this, it would be better to use a constant string or something like that.

(It's somewhat unfortunate that typescript doesn't have reasonable support for enums with different variants, which would be ideal here)

Reply:
I have addressed this by using a string NA instead.